### ISSUE TITLE
opus files are downloaded as octet-stream

### ISSUE THREAD

**Issue by masewo** [tus/tusd#1194]
**Describe the bug**
Files with content type `audio/ogg; codecs=opus` are downloaded as `application/octet-stream`.
The content type does not pass the regex check in the `unrounted_handler.go`.

**To Reproduce**
1. Upload an opus file
2. Download this file
3. See content type

**Expected behavior**
The content type is `audio/ogg; codecs=opus`.

**Setup details**
Please provide following details, if applicable to your situation:
- Operating System: Windows
- Used tusd version: 2.5.0
- Used tusd data storage: disk store
- Used tusd configuration: Go
- Used tus client library: Uppy.js

**Comment by Acconut** [comment]
That's correct. Additional parameters in the media type are currently not allowed. Would you be interested in a PR for extending the regular expression to allow the syntax specified in https://httpwg.org/specs/rfc9110.html#media.type?

Alternatively, we could also consider moving away from the regular expression entirely and instead use Go's builtin media type parser: https://pkg.go.dev/mime#ParseMediaType. The benefit is that we do not have to maintain a complex regular expression that might be incompatible with the actual specs.

If the parsing fails, we would use `application/octet-stream`. Otherwise, we set `Content-Disposition` depending on whether the media type is allowed to be displayed inline or not.

What do you think?

**Comment by Acconut** [comment]
> I did not have good experiences recently with go's built-in mime package since it relies on information of the OS where it is installed on (causing issue for me on Alpine systems).

That's surprising to me as I would expect media parsing to not load OS resources since it's just parsing a string. However, I haven't tested this on my own yet. Nevertheless, I would like to explore that route more as I also mentioned in a comment in your PR.

**Comment by Acconut** [comment]
Fixed in https://github.com/tus/tusd/pull/1253.
